File: frontend/types/formulario-606.ts
// Campos exactos del Formato 606 de la DGII
export interface Registro606 {
  id: string;
  rnc_proveedor: string;         // RNC o cédula del proveedor
  tipo_id: "1" | "2" | "3";     // 1=RNC, 2=Cédula, 3=Pasaporte
  ncf: string;                   // Número de comprobante fiscal
  ncf_modificado?: string;       // NCF nota de crédito/débito (si aplica)
  tipo_compra: string;           // Tipo de bienes/servicios (01-16)
  fecha_comprobante: string;     // YYYYMMDD
  fecha_pago: string;            // YYYYMMDD
  monto_bienes: number;
  monto_servicios: number;
  total_facturado: number;
  itbis_facturado: number;
  itbis_retenido: number;
  itbis_proporcional: number;
  costo_servicio: number;
  bienes_capital: number;
  gastos_sujetos: number;
  // Estado interno (no va en el TXT)
  valido: boolean;
  errores?: string[];
}

export interface Declaracion606 {
  id: string;
  periodo: string;               // YYYYMM
  rnc_empresa: string;
  registros: Registro606[];
  estado: "borrador" | "validado" | "enviado";
  version_formulario: string;
  creado_en: string;
  actualizado_en: string;
}

// Respuesta del backend al extraer datos
export interface ExtractResponse {
  registros: Registro606[];
  advertencias: string[];
  archivos_procesados: number;
}

File: backend/models/formulario_606.py
from pydantic import BaseModel, Field
from typing import Optional


class Registro606(BaseModel):
    """Campos exactos del Formato 606 de la DGII"""
    rnc_proveedor: str = Field(..., description="RNC o cédula del proveedor")
    tipo_id: str = Field(..., description="1=RNC, 2=Cédula, 3=Pasaporte")
    ncf: str = Field(..., description="Número de comprobante fiscal")
    ncf_modificado: Optional[str] = Field(None, description="NCF nota crédito/débito")
    tipo_compra: str = Field(..., description="Tipo de bienes/servicios 01-16")
    fecha_comprobante: str = Field(..., description="YYYYMMDD")
    fecha_pago: str = Field(..., description="YYYYMMDD")
    monto_bienes: float = Field(default=0.0)
    monto_servicios: float = Field(default=0.0)
    total_facturado: float
    itbis_facturado: float = Field(default=0.0)
    itbis_retenido: float = Field(default=0.0)
    itbis_proporcional: float = Field(default=0.0)
    costo_servicio: float = Field(default=0.0)
    bienes_capital: float = Field(default=0.0)
    gastos_sujetos: float = Field(default=0.0)


class ExtractResponse(BaseModel):
    registros: list[Registro606]
    advertencias: list[str] = []
    archivos_procesados: int
